invenio_records_rest/serializers/datacite.py

```diff
@@ -75,7 +79,31 @@ def serialize_oaipmh(self, pid, record):
             if isinstance(record['_source'], Record) \
             else self.transform_search_hit(pid, record)
 
-        return schema31.dump_etree(obj)
+        return self.schema.dump_etree(obj)
+
+
+class DataCite31Serializer(BaseDataCiteSerializer):
+    """Marshmallow DataCite serializer v3.1 for records.
+
+    Note: This serializer is not suitable for serializing large number of
+    records.
+    """
+
+    """Class variable to define which schema use."""
+    schema = schema31
+    version = '3.1'
+
+
+class DataCite40Serializer(BaseDataCiteSerializer):
+    """Marshmallow DataCite serializer v4.0 for records.
+
+    Note: This serializer is not suitable for serializing large number of
+    records.
+    """
+
+    """Class variable to define which schema use."""
+    schema = schema40
+    version = '4.0'
 
 
 class OAIDataCiteSerializer(object):
```

Comment:
Here, I left the parameter "v31" in the `__init__` method for backward compatibility. But it may be better to change it as it may no longer be the v31 of Datacite...